### PR TITLE
Added Fast Inverse Square Root

### PR DESCRIPTION
**Pull request by aryanshb**
### Describe your change:

* [X] Add an algorithm?
* [ ] Fix a bug or typo in an existing algorithm?
* [ ] Add or change doctests? -- Note: Please avoid changing both code and tests in a single pull request.
* [ ] Documentation change?

### Checklist:
* [X] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md).
* [X] This pull request is all my own work -- I have not plagiarized.
* [X] I know that pull requests will not be merged if they fail the automated tests.
* [X] This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
* [X] All new Python files are placed inside an existing directory.
* [X] All filenames are in all lowercase characters with no spaces or dashes.
* [X] All functions and variable names follow Python naming conventions.
* [X] All function parameters and return values are annotated with Python [type hints](https://docs.python.org/3/library/typing.html).
* [X] All functions have [doctests](https://docs.python.org/3/library/doctest.html) that pass the automated testing.
* [X] All new algorithms include at least one URL that points to Wikipedia or another similar explanation.
* [ ] If this pull request resolves one or more open issues then the description above includes the issue number(s) with a [closing keyword](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue): "Fixes #ISSUE-NUMBER".

### CODE REVIEW

File: maths/fast_inverse_sqrt.py [hunk 1]
```diff
@@ -0,0 +1,35 @@
+"""
+Fast inverse square root.
+Reference: https://en.wikipedia.org/wiki/Fast_inverse_square_root
+"""
+
+import struct
+
+
+def fastInvSqrt(x: float) -> float:
```
Comment (cclauss) [inline maths/fast_inverse_sqrt.py:9]:
```suggestion
def fast_inverse_sqrt(x: float) -> float:
```

File: maths/fast_inverse_sqrt.py [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+
+    :param float x: The input number for which to calculate the inverse square root.
+    :return float: The fast inverse square root of the input number.
+
+    Example:
+    >>> fastInvSqrt(10)
+    0.3156857923527257
+
+    >>> fastInvSqrt(4)
+    0.49915357479239103
```
Comment (cclauss) [inline maths/fast_inverse_sqrt.py:24]:
```suggestion
    >>> fast_inverse_sqrt(10)
    0.3156857923527257
    >>> fast_inverse_sqrt(4)
    0.49915357479239103
    >>> fast_inverse_sqrt(0)
    ???
    >>> fast_inverse_sqrt(-4)
    ???
    >>> fast_inverse_sqrt(4.1)
    ???
    >>> from math import sqrt
    >>> all(fast_inverse_sqrt(i) == sqrt(i) for i in range(-10, 10))
    True
```

**Review by cclauss — CHANGES_REQUESTED** [review]
Add some doctests that break things.

**Comment by aryanshb** [comment]
The test may/may not fail due to the approximation of answer, which is the essence of this method.

Pls review for any additional change.

**Comment by cclauss** [comment]
As discussed in CONTRIBUTING.md you can run the tests on your own computer with:
`python3 -m doctest -v maths/fast_inverse_sqrt.py`

**Comment by aryanshb** [comment]
> As discussed in CONTRIBUTING.md you can run the tests on your own computer with: `python3 -m doctest -v maths/fast_inverse_sqrt.py`

The doctests work perfectly on my PC, even before i committed. In the build fail details, `fetch_anime_and_play.py` seem to be failing tests.

image

**Review by cclauss — APPROVED** [review]
Final edits: https://github.com/TheAlgorithms/Python/pull/11054/commits/aeae0a809aa5cc07321f1f0cdddaf77b3de17006

**Pull request merged**